### ISSUE TITLE
MSG ARRIVED Sporadic

### ISSUE THREAD

**Issue by Chluz** [sblantipodi/smart_watch_winder#2]
Hi sblantipodi, 

great project ! I'm integrating this in a 3D print, will share pictures when I am done. 
I've been testing with only 1 d1 mini for the time being, and I'm getting strange behaviour 

I recompiled the code, initially as it was, and then I noticed there was a dependency on a CLIMATE device -- so I rewrote that bit to finally get the watchwinder to publish its "info" to the mqtt server). 

Now though, when I make changes in homeassistant, I see the change populating in mqtt, but the d1 mini is not registering the change. 
Having activated the debug mode and connected the serial interface whilst it was running, I see that the state (rotation number) of the winder is being updated every few seconds, but that mqtt commands are only detected sporadically (indicated by the  MSG ARRIVED on the serial debug interface). 

Do you have any idea what the issue could be ?
I could not figure out the code for the MQTT subscription updates in your bootstrapper (i never program cpp normally)

Thanks in advance,

**Comment by Chluz** [comment]
Hi, so after much testing, I finally found out that the power supply seems to have a big impact on on the D1 mini's ability to receive messages from the subscribe thread. 
Seems like it is working well now with a step down 2Amp power supply from 12V to 5V, powered by the 5V pin. 

I have seen a lot of posts about issues with the subscribe function in the PubSubClient issues thread, I suspect in many cases it might be the same issue I was having. 

So everything is well with your code (after the modifications I made to remove the dependency on the CLIMATE time posting). I will close this issue.
